examples/mini-hlint/src/Main.hs

```diff
@@ -107,15 +124,20 @@ main = do
   case args of
     [file] -> do
       s <- readFile' file
-      let flags = defaultDynFlags fakeSettings fakeLlvmConfig
-      case parse file flags s of
-        PFailed s ->
-          report flags $ snd (getMessages s flags)
-        POk s m -> do
-          let (warns, errs) = getMessages s flags
-          report flags warns
-          report flags errs
-          when (null errs) $ analyzeModule flags m
+      flags <-
+        parsePragmasIntoDynFlags
+          (defaultDynFlags fakeSettings fakeLlvmConfig) file s
+      whenJust flags
```

Comment:
`whenJust flags $ \flags ->` would be the normal way to use `whenJust` here.

Reply:
Will fix

Reply:
An opportunity for a new `hlint` hint here?

Reply:
Perhaps... but it's more subjective I guess.